internal/server/data/data.go

```diff
@@ -271,53 +310,44 @@ func Count[T models.Modelable](db *gorm.DB, selectors ...SelectorFunc) (int64, e
 	return count, nil
 }
 
-var infraProviderCache *models.Provider
-
 // InfraProvider is a lazy-loaded cached reference to the infra provider. The
 // cache lasts for the entire lifetime of the process, so any test or test
 // helper that calls InfraProvider must call InvalidateCache to clean up.
 func InfraProvider(db *gorm.DB) *models.Provider {
-	if infraProviderCache == nil {
-		infra, err := get[models.Provider](db, ByProviderKind(models.ProviderKindInfra))
-		if err != nil {
-			if errors.Is(err, internal.ErrNotFound) {
-				p := &models.Provider{Name: models.InternalInfraProviderName, Kind: models.ProviderKindInfra}
-				if err := add(db, p); err != nil {
-					logging.L.Panic().Err(err).Msg("failed to create infra provider")
-				}
-				return p
+	org := OrgFromContext(db.Statement.Context)
+	infra, err := get[models.Provider](db, ByProviderKind(models.ProviderKindInfra), ByOrgID(org.ID))
```

Comment:
We will have to think about re-visiting this at some point to add the caching back imo, fine for now though

Reply:
Ya, we should look at this again. I think we should try and cache these things and the default org on `Server` , so that we can pass it via `RequestContext`.